cmd/plugin/commands/versionlint/main.go

```diff
@@ -0,0 +1,173 @@
+)
+
+// CreateCommand creates and returns this cobra subcommand
+func CreateCommand(flags *genericclioptions.ConfigFlags) *cobra.Command {
+	var opts *lintOptions
+	cmd := &cobra.Command{
+		Use:   "version-lint",
+		Short: "Inspect kubernetes resources to see if they need to be changed for a new ingress-nginx version",
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("Checking ingresses...")
+			err := ingresses(*opts)
+			if err != nil {
+				util.PrintError(err)
+			}
+			fmt.Println("Checking deployments...")
+			err = deployments(*opts)
+			if err != nil {
+				util.PrintError(err)
+			}
+		},
+	}
+
+	opts = addCommonOptions(flags, cmd)
+
+	cmd.AddCommand(createSubcommand(flags, []string{"ingresses", "ingress", "ing"}, "Check ingresses for possible issues", ingresses))
+	cmd.AddCommand(createSubcommand(flags, []string{"deployments", "deployment", "dep"}, "Check deployments for possible issues", deployments))
+
+	return cmd
+}
+
+func createSubcommand(flags *genericclioptions.ConfigFlags, names []string, short string, f func(opts lintOptions) error) *cobra.Command {
+	var opts *lintOptions
+	cmd := &cobra.Command{
+		Use:     names[0],
+		Aliases: names[1:],
+		Short:   short,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			util.PrintError(f(*opts))
+			return nil
+		},
+	}
+
+	opts = addCommonOptions(flags, cmd)
+
+	return cmd
+}
+
+func addCommonOptions(flags *genericclioptions.ConfigFlags, cmd *cobra.Command) *lintOptions {
+	out := lintOptions{
+		flags: flags,
+	}
+	cmd.Flags().BoolVar(&out.allNamespaces, "all-namespaces", false, "Check ingresses in all namespaces")
+	cmd.Flags().BoolVar(&out.showAll, "show-all", false, "Show all ingresses, not just the ones with problems")
+	return &out
+}
+
+type lintOptions struct {
+	flags         *genericclioptions.ConfigFlags
+	allNamespaces bool
+	showAll       bool
+}
+
+type lint interface {
+	Check(obj kmeta.Object) bool
+	Message() string
+}
+
+func checkObjectArray(lints []lint, objects []kmeta.Object, showAll bool) {
+	for _, obj := range objects {
+		messages := make([]string, 0)
+		for _, lint := range lints {
+			if lint.Check(obj) {
+				messages = append(messages, lint.Message())
+			}
+		}
+		if len(messages) == 0 {
+			if showAll {
+				fmt.Printf("✓ %v/%v\n", obj.GetNamespace(), obj.GetName())
+			}
+		} else {
+			fmt.Printf("✗ %v/%v\n", obj.GetNamespace(), obj.GetName())
+			for _, msg := range messages {
+				fmt.Printf("  - %v\n", msg)
+			}
+			fmt.Println("")
+		}
+	}
+}
+
+func ingresses(opts lintOptions) error {
+	var ings []v1beta1.Ingress
+	var err error
+	if opts.allNamespaces {
+		ings, err = request.GetIngressDefinitions(opts.flags, "")
+	} else {
+		ings, err = request.GetIngressDefinitions(opts.flags, util.GetNamespace(opts.flags))
+	}
+	if err != nil {
+		return err
+	}
+
+	var iLints []lints.IngressLint = lints.GetIngressLints()
+	genericLints := make([]lint, len(iLints))
+	for i := range iLints {
```

Comment:
`copy(genericLints, lints.GetIngressLints())` ?

Reply:
They have different types. genericLints is []lint, as opposed to []lints.IngressLint. Manually copying them is the only way to convert between the two.